examples/mocks/nested/nested.as.ts

```diff
@@ -0,0 +1,7 @@
+async function plugin (fastify: any, opts: any) {
+  fastify.get('/nested', async (req: any, reply: any) => {
```

Comment:
Yep, we are going to provide the store instance to the plugin.

Reply:
This is the overall idea:

factoryCreateMock() {
    store,
    mocksRegistry
} -> fastifyPlugin route with store

Reply:
Nice. Simply currently I can't use the plugins from fs, because their calls aren't recorded. 

But then, you might want to attack mockRegistry design which is very basic. 
Currently, your plugins aren't in mockRegistry, which has some downsides. Some day, somehow, it should be possible to reset mocksRegistry or unregister a plugin handler from HTTP level etc.
You can also burn mockRegistry or rename it to httpMocksRegistry etc.

Reply:
Sure thing. But for now I think we can merge this PR and focus on the registerMockFactory function.